pkg/cmd/opts/install.go

```diff
@@ -777,10 +777,51 @@ func (o *CommonOptions) InstallMinishift() error {
 
 // InstallGcloud installs gcloud cli
 func (o *CommonOptions) InstallGcloud() error {
-	if runtime.GOOS != "darwin" || o.NoBrew {
-		return errors.New("please install missing gcloud sdk - see https://cloud.google.com/sdk/downloads#interactive")
+	binDir, err := util.JXBinLocation()
+	binary := "gcloud"
+	if err != nil {
+		return err
+	}
+	// Check is binary exists in JX_HOME/bin
+	fullBinaryPath := filepath.Join(binDir, "gcloud")
+	exists, err := util.FileExists(fullBinaryPath)
+	if err != nil {
+		return errors.Wrapf(err, "unable to verify if binary exists")
+	}
+	if exists {
+		log.Logger().Debugf("binary %s already exists", fullBinaryPath)
+		return nil
+	}
+	// get the stable jx supported version of gcloud to be installed
+	versionResolver, err := o.GetVersionResolver()
+	if err != nil {
+		return err
+	}
+	stableVersion, err := versionResolver.StableVersionNumber(versionstream.KindPackage, "gcloud")
+	if err != nil {
+		return err
+	}
+	clientURL := fmt.Sprintf("https://storage.googleapis.com/cloud-sdk-release/google-cloud-sdk-%s-%s-x86.tar.gz", stableVersion, runtime.GOOS)
+	if strings.Contains(runtime.GOARCH, "64") {
+		clientURL = fmt.Sprintf("https://storage.googleapis.com/cloud-sdk-release/google-cloud-sdk-%s-%s-x86_64.tar.gz", stableVersion, runtime.GOOS)
 	}
-	return o.RunCommand("brew", "cask", "install", "google-cloud-sdk")
+	
+	fileName, flag, err := packages.ShouldInstallBinary(binary)
+	if err != nil || !flag {
+		return err
+	}
+	fullPath := filepath.Join(binDir, fileName)
+	tarFile := fullPath + ".tgz"
+	err = packages.DownloadFile(clientURL, tarFile)
+	if err != nil {
+		return err
+	}
+	err = util.UnTargz(tarFile, binDir, []string{binary, fileName})
```

Comment:
Please also remove the un-used files and tar-ball after we've copied the binary.